sympy/core/numbers.py

```diff
@@ -2469,6 +2471,14 @@ def _eval_power(self, expt):
             return S.NaN
         if expt is S.ComplexInfinity:
             return S.NaN
+        if expt.is_real is False:
+            expt_real = re(expt)
```

Comment:
I don't know if that is going to be expensive. How can i measure its expensiveness? Should i use the added time to the tests running of `test_numbers.py` as its measure? Perhaps checking whether `real-part` of `expt <, >, = 0`  is necessary. Though in the latest commit, i have made included it in `expt.is_number` to avoid this operation comparison for `non-numbers`(symbols).